app/api/chat/route.ts

```diff
@@ -0,0 +1,38 @@
+import { Configuration, OpenAIApi } from "openai-edge";
+import { OpenAIStream, StreamingTextResponse } from "ai";
+
+export const runtime = "edge";
+
+const apiConfig = new Configuration({
+  apiKey: "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",
+});
+const openai = new OpenAIApi(apiConfig);
+
+export async function POST(req: Request) {
+  // Extract the `messages` from the body of the request
+  const { messages } = await req.json();
+
+  // Request the OpenAI API for the response based on the prompt
+  const response = await openai.createChatCompletion({
```

Comment:
This API can be reused to a separate function should be created.

const getChatApi = async(message) => {
// your API call
}